interfaces/devdoc/buffer_requirements.md

```diff
@@ -0,0 +1,248 @@
+BUFFER Requirements
```

Comment:
>Requirements [](start = 7, length = 12)

in any way, this file should not be under "interfaces/devdoc".

Because buffer is not an interface... that's why.

Reply:
+1 belongs in the common folder rather than interfaces folder

---
In reply to: [450451974](https://github.com/Azure/azure-c-pal/pull/7#discussion_r450451974) [](ancestors = 450451974)

Reply:
Same for the headers that belong in the common inc folder

---
In reply to: [450474388](https://github.com/Azure/azure-c-pal/pull/7#discussion_r450474388) [](ancestors = 450474388,450451974)

Reply:
I moved buffer and constbuffer into common. Any others that need to be moved?


---
In reply to: [450474582](https://github.com/Azure/azure-c-pal/pull/7#discussion_r450474582) [](ancestors = 450474582,450474388,450451974)